### PR TITLE
feat: add flag to auto-delete oldest theme during stencil push

### PR DESCRIPTION
**Pull request by bc-NikitaPuzanenko**
#### What?

Adding --delete and -d flags to stencil push. This flag will automatically delete the oldest private theme if the upload limit is reached.

#### Tickets / Documentation

https://github.com/bigcommerce/stencil-cli/issues/429

#### Screenshots (if appropriate)

Before:
before

After:
after-2

### CODE REVIEW

File: lib/stencil-push.utils.js [hunk 1]
```diff
@@ -152,6 +151,14 @@ utils.promptUserToDeleteThemesIfNecessary = (options, callback) => {
         return async.nextTick(callback.bind(null, null, options));
     }
 
+    if (options.deleteOldest) {
+        const oldestTheme =  options.themes
+        .filter(theme => theme.is_private)
+        .map(theme => ({name:theme.name, uuid: theme.uuid, updated_at : theme.updated_at = new Date(theme.updated_at).valueOf()}))
```
Comment (lord2800) [inline lib/stencil-push.utils.js]: Is the `name` here just for inspection convenience? It looks like you're only using the `uuid` and the `updated_at`.

Also, is the `theme.updated_at = ` part necessary?

Comment (bc-NikitaPuzanenko) [reply lib/stencil-push.utils.js]: Yeah the name can likely be removed, I used it to check against what is in the control panel to make sure it is getting the oldest theme.

Will also remove the `theme.updated_at = ` assignment, not sure why I wrote that.

**Comment by LordZardeck** [comment]
This would be immensely helpful and would simplify and reduce the amount of work needed to get this working in CI/CD. @lord2800 Any chance this will be merged and released soon?

**Comment by lord2800** [comment]
cc @junedkazi @mattolson who maintain the releases for stencil-cli.

File: lib/stencil-push.utils.js [hunk 2]
```diff
@@ -152,6 +151,14 @@ utils.promptUserToDeleteThemesIfNecessary = (options, callback) => {
         return async.nextTick(callback.bind(null, null, options));
     }
 
+    if (options.deleteOldest) {
+        const oldestTheme =  options.themes
+        .filter(theme => theme.is_private)
```
Comment (junedkazi) [inline lib/stencil-push.utils.js]: We should also filter the active theme here.

**Comment by mattolson** [comment]
@lord2800 as soon as this is approved and merged, the storefront team can cut a release.

**Comment by 0dp** [comment]
@mattolson 

It looks like `--delete` haven't made it to npm yet.
```
$stencil -V
1.21.4
```

```
stencil push --delete
error: unknown option `--delete'
```

**Comment by carsonreinke** [comment]
@0dp it hasn't been released yet

**Comment by 0dp** [comment]
But the docs are live 🙈

**Comment by christophersjchow** [comment]
Any updates on when this will be released?

**Comment by junedkazi** [comment]
@christophersjchow it has been released a while back.

**Pull request merged**